<launch>
  <!-- Global parameters (for PathFootprint in tier4_planning_rviz_plugin) -->
  <!-- Do not add "group" in order to propagate global parameters -->
  <include file="$(find-pkg-share global_parameter_loader)/launch/global_params.launch.py">
    <arg name="vehicle_model" value="ymc_golfcart_m0"/>
  </include>

  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <!-- <arg name="input_path_topic" default="input/path"/> -->
  <arg name="input_path_topic" default="raw_path"/>
  <arg name="output_path_topic" default="output/path"/>
  <arg name="is_showing_debug_info" default="false"/>
  <arg name="param_path" default="$(find-pkg-share static_path_smoother)/config/static_path_smoother.param.yaml"/>

  <node pkg="static_path_smoother" exec="static_path_smoother_node" name="static_path_smoother_optimization" output="screen">
    <remap from="~/input/objects" to="$(var input_objects)"/>
    <remap from="~/input/path" to="$(var input_path_topic)"/>
    <remap from="~/output/path" to="$(var output_path_topic)"/>

    <!-- <param from="$(var param_path)"/> -->
    <param name="is_showing_debug_info" value="$(var is_showing_debug_info)"/>

    <param from="$(find-pkg-share planning_launch)/config/scenario_planning/lane_driving/motion_planning/obstacle_avoidance_planner/obstacle_avoidance_planner.param.yaml"/>
    <param from="$(find-pkg-share planning_launch)/config/scenario_planning/common/common.param.yaml"/>
    <param from="$(find-pkg-share planning_launch)/config/scenario_planning/common/nearest_search.param.yaml"/>
  </node>
</launch>
